context.go

```diff
@@ -0,0 +1,288 @@
+package main
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/viper"
+	yaml "gopkg.in/yaml.v2"
+
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"text/tabwriter"
+
+	"github.com/fnproject/cli/config"
+	"github.com/urfave/cli"
+)
+
+var contextsPath = filepath.Join(config.RootConfigPathName, config.ContextsPathName)
+var defaultContextPath string
+
+func contextCmd() cli.Command {
+	return cli.Command{
+		Name:  "context",
+		Usage: "manage context",
+		Subcommands: []cli.Command{
+			{
+				Name:      "create",
+				Aliases:   []string{"c"},
+				Usage:     "create a new context",
+				ArgsUsage: "<context>",
+				Action:    create,
+				Flags: []cli.Flag{
+					cli.StringFlag{
+						Name:  "provider",
+						Usage: "context provider",
+					},
+					cli.StringFlag{
+						Name:  "api-url",
+						Usage: "context api url",
+					},
+					cli.StringFlag{
+						Name:  "registry",
+						Usage: "context registry",
+					},
+				},
+			},
+			{
+				Name:      "delete",
+				Aliases:   []string{"d"},
+				Usage:     "delete a context",
+				ArgsUsage: "<context>",
+				Action:    delete,
+			},
+			{
+				Name:    "list",
+				Aliases: []string{"l"},
+				Usage:   "list contexts",
+				Action:  list,
+			},
+			{
+				Name:      "set",
+				Aliases:   []string{"s"},
+				Usage:     "set context for future invocations",
+				ArgsUsage: "<context>",
+				Action:    set,
+			},
+			{
+				Name:   "unset",
+				Usage:  "unset current-context",
+				Action: unset,
+			},
+		},
+	}
+}
+
+func create(c *cli.Context) error {
+	context := c.Args().Get(0)
+
+	re := regexp.MustCompile("[^a-zA-Z0-9_-]+")
+
+	for range re.FindAllString(context, -1) {
+		fmt.Fprintf(os.Stderr, "please enter a context name with ASCII characters only \n")
+		os.Exit(1)
+	}
+
+	provider := config.DefaultProvider
+	if cProvider := c.String("provider"); cProvider != "" {
+		provider = cProvider
+	}
+
+	apiUrl := config.DefaultLocalApiUrl
+	if cApiUrl := c.String("api-url"); cApiUrl != "" {
+		apiUrl = cApiUrl
+	}
+
+	registry := ""
+	if cRegistry := c.String("registry"); cRegistry != "" {
+		registry = cRegistry
+	}
+
+	if check, err := checkContextFileExists(context); check {
+		if err != nil {
+			return fmt.Errorf("%v", err)
+		}
+		return errors.New("context already exists")
+
+	}
+	path, err := createFilePath(context)
+	if err != nil {
+		return fmt.Errorf("%v", err)
+	}
+	_, err = os.Create(path)
+	if err != nil {
+		return fmt.Errorf("%v", err)
+	}
```

Comment:
It probably doesn't matter since this program will exit , but this will leak the File object returned by `os.Create`.

But `config.WriteYamlFile` below will perform the file truncation and create as well, so this call to `os.Create` may not be necessary?